### DISCUSSION TITLE
MQTT limit_nonpersistent_absolute

### DISCUSSION THREAD

**Discussion by usekevin1988** [tbnobody/OpenDTU#431] [Q&A] [answered]
Hey Community.

Weiss jemand, wie oft ich den HM-600 per MQTT /cmd/limit_nonpersistent_absolute regeln kann? Sind alle 5 Sekunden zu oft?

Will eine Nulleinspeisung umsetzen und muss dafür natürlich häufig regeln

EDIT:// Wenn ich alle 5 Sekunden einen Wert per MQTT raushaue, aktualisiert "Data Age" sich nur noch Sporadisch. Wenn ich es auf 10 Sekunden erhöhe läuft es mehr oder weniger flüssig

**Comment by dAjaY85** [comment]
Moin, 
meines Wissens, sendet der DTU alle 30 s Nachrichten an den hoymiles.

Ich sende das Limit alle 10 Minuten, könnte mit Sicherheit auch reduzieren, hab aber noch keinen Bedarf dazu.

Könnte mir vorstellen, dass mit 60 s problemlos laufen sollte. Weiß leider nicht, wie das Innenleben im hoymiles auf die Änderungen reagiert.

**Comment by hartkopp** [comment, marked as answer]
Moin,
ich prüfe alle 5 Sekunden die Einspeise-Leistung von meinem Tasmota (Smart Meter WLAN Knoten).
Und wenn sich eine Änderung ergibt (insbesondere eine ungewollte Einspeisung), dann regele ich alle 5 Sekunden nach.
https://github.com/hartkopp/zerofeed/blob/main/zerofeed.sh#L212

Ansonsten versuche ich den Verbrauch in einem bestimmten Fenster zu halten, um seltener regeln zu müssen.

Die erfolgreiche Übernahme des Limitwertes wird in einer separaten Funktion abgewartet.

Siehe https://github.com/hartkopp/zerofeed

**Comment by daebikelinz** [comment]
Hallo zusammen,

hat sich jemand schon über die Lebensdauer des RAM/EEPROM bei den Schreibzyklen im 5 Sekunden Intervall Gedanken gemacht?
Mich würde interessieren ob das "limit_nonpersistent_absolute" auch wirklich in den RAM geschrieben wird und nicht im EEPROM.
Habe eine gewisse "Angst" den Inverter zu bricken wenn man alle 5 oder 10 Sekunden einen Wert schreibt.
8h/24h * 60 min * 6 Schreibzyklen/min = 2880 Schreibzyklen pro Tag = 1 Mio / Jahr.
Hat jemand Infos hierzu?

**Reply by usekevin1988** [reply]
mit nonpersistant quasi unendlich. Mit persistant
 schreiben sieht das ganze schon anders aus, weil es dann ebend nicht mehr nur in den Ram geschrieben wird.

Da Hoymiles selbst mit originaler DTU auch eine nulleinspeisung anbietet, sehe ich da kein problem bzgl. Haltbarkeit

5sek Intervall funktioniert aber auch nicht. da kommt der WR nicht mehr mit. 10 sekunden müssens schon sein

siehe: https://github.com/tbnobody/OpenDTU/discussions/502

**Reply by hartkopp** [reply]
Hallo @usekevin1988 ,

danke für die Bestätigung mit dem non-persistent Verfahren, dass Hoymiles das auch so umsetzt. Das ist beruhigend ;-)

Ich habe sowohl in der Aktualisierung der OpenDTU-Website also auch in meinem Script eine Aktualisierungsrate von 5 Sekunden (das Script ist etwas langsamer, weil es immer noch das Smartmeter abfragt).
Nach der Übergabe eines neuen Wertes an den Limiter ist der Effekt schon nach 1-2 Sekunden im Smartmeter sichtbar. Wofür braucht es bei Dir 10 Sekunden?

Ich frage übrigens auch ab, ob der Befehl zum Setzen des Limiters auch sauber durchgeführt wurde:
https://github.com/hartkopp/zerofeed/blob/main/zerofeed-dbg.sh#L106

Seitdem läuft das Setzen der Werte sehr robust.

**Reply by daebikelinz** [reply]
> mit nonpersistant quasi unendlich. Mit persistant schreiben sieht das ganze schon anders aus, weil es dann ebend nicht mehr nur in den Ram geschrieben wird.
> 
> Da Hoymiles selbst mit originaler DTU auch eine nulleinspeisung anbietet, sehe ich da kein problem bzgl. Haltbarkeit
> 
> 5sek Intervall funktioniert aber auch nicht. da kommt der WR nicht mehr mit. 10 sekunden müssens schon sein
> 
> siehe: #502

Das mit dem Hinweis das Hoymiles das genauso macht ist beruhigend und macht echt Sinn. Danke @usekevin1988 !
Bei mir bleibt leider noch ein gewisses Restrisiko weil ich einen TSUN M350 benutze und keinen Hoymiles...

Werde wohl auf gut Glück hier mal "Pionierarbeit" leisten und dabei die Schreibzyklen mit tracken. 

@hartkopp So mache ich das auch. Erst sicher gehen ob der Inverter die neue Leistung schon ausgibt bevor ein neuer Wert reinflattert.

**Reply by usekevin1988** [reply]
> Hallo @usekevin1988 ,
> 
> danke für die Bestätigung mit dem non-persistent Verfahren, dass Hoymiles das auch so umsetzt. Das ist beruhigend ;-)
> 
> Ich habe sowohl in der Aktualisierung der OpenDTU-Website also auch in meinem Script eine Aktualisierungsrate von 5 Sekunden (das Script ist etwas langsamer, weil es immer noch das Smartmeter abfragt). Nach der Übergabe eines neuen Wertes an den Limiter ist der Effekt schon nach 1-2 Sekunden im Smartmeter sichtbar. Wofür braucht es bei Dir 10 Sekunden?
> 
> Ich frage übrigens auch ab, ob der Befehl zum Setzen des Limiters auch sauber durchgeführt wurde: https://github.com/hartkopp/zerofeed/blob/main/zerofeed-dbg.sh#L106
> 
> Seitdem läuft das Setzen der Werte sehr robust.

Wenn ich bei meiner NodeRed umsetzung auf unter 10 Sekunden gehe, (ich habs aber auch nicht mit 6-9 Sekunden probiert) kommt der WR nicht mehr sauber hinterher. Ich frage aber auch nicht ab, ob das limit schon gesetzt wurde, bzw beziehe es nicht mit ein in der funktion. Wollte es möglichst schlicht halten

**Reply by hartkopp** [reply]
> Wenn ich bei meiner NodeRed umsetzung auf unter 10 Sekunden gehe, (ich habs aber auch nicht mit 6-9 Sekunden probiert) kommt der WR nicht mehr sauber hinterher. Ich frage aber auch nicht ab, ob das limit schon gesetzt wurde, bzw beziehe es nicht mit ein in der funktion. Wollte es möglichst schlicht halten

Das hatte ich auch erst. Wenn man sich die Reaktion des Interverters aber anschaut, dann gibt es von 1 Sek bis zu 3 oder 4 Sek alles mögliche an Antwortzeiten. Erst nach der Prüfung des Limiter-Kommandos funktionierte das Script richtig stabil.

**Reply by usekevin1988** [reply]
das kann durchaus sein. Aber für mich reichen alle 10 Sekunden. Optimieren kann man auch meinen NodeRed ansatz bestimmt noch.
